Add verbatim and embed engine (#369)
rstudio/rmarkdown-cookbook: 05-formatting.Rmd

Typically we write code chunks and inline expressions that we want to be parsed and evaluated by **knitr**. However, if you are trying to write a tutorial on using **knitr**, you may need to generate a verbatim code chunk or inline expression that is _not_ parsed by **knitr**, and we want to display the content of the chunk header. 

> The `verbatim` engine will take any type of R Markdown valid content and output it as-is in the resulting document. By adding one more backtick in the upper layer than the inside content, a whole code chunk including the chunk header can be included. 

Reviewer: ```suggestion
The `verbatim` engine can take any R Markdown content and output it as-is in the resulting document. By using more backticks on the outer fences than the backticks in the inner content, a whole code chunk including the chunk header can be included in a `verbatim` chunk. For example,
```

`````{verbatim}
````{verbatim}
```{r, eval=TRUE}
1 + 1
```
````
`````

will be rendered as: 

````{verbatim}
```{r, eval=TRUE}
1 + 1
```
````

> in the output. You have to use at least N+1 backticks to wrap up N backticks, so four backticks here for the chunk header with the `verbatim` engine. 

Reviewer: ```suggestion
in the output. You have to use at least `N+1` backticks to wrap up `N` backticks. We are using four backticks here for the `verbatim` chunk because the inner chunk header contains three backticks.
```

> The content of the chunk will be placed in a fenced code block with class `default`, meaning not specific highlighting applied. You can set the `lang` chunk option to use a different language. For example, 

Reviewer: ```suggestion
The content of the `verbatim` chunk will be placed in a fenced code block with the class `default`, meaning no syntax highlighting will be applied. You can set the `lang` chunk option to use a different syntax highlighting language name. For example, 
```

`````{verbatim}
````{verbatim, lang = "markdown"}
We can output arbitrary content **verbatim**.

```{r}
1 + 1
```

The content can contain inline code like
`r pi * 5^2`, too.
````
`````

> will result in the markdown source file as 

Reviewer: ```suggestion
will result in the Markdown source file as 
```

`````{verbatim, lang = "markdown"}
````markdown
We can output arbitrary content **verbatim**.

```{r}
1 + 1
```

The content can contain inline code like
`r pi * 5^2`, too.
````
`````

> which will apply highlighting for markdown language on the code chunk in the output.

Reviewer: ```suggestion
The verbatim content will be syntax highlighted using the `markdown` language in the output.
```

> If the content you want to include verbatim is already in a file, you can use the `embed` engine instead. Here is an example of the CSS we are using to create boxes in this book: 

Reviewer: ```suggestion
If the content you want to include verbatim is in a file, you can use the `embed` engine instead. Here is an example of the CSS code from Section \@ref(block-shaded):
```
